test/ably/restappstats_test.py

```diff
@@ -28,313 +29,49 @@ def setUpClass(cls):
-
-        self.assertTrue(stats_pages.has_next)
-        stats_pages = stats_pages.get_next()
-        stats_page = stats_pages.current
-
-        self.assertEqual(1, len(stats_page), "Expected 1 record")
-        self.assertEqual(60, stats_page[0].inbound.all.all.count, "Expected 60 messages")
-
-        self.assertTrue(stats_pages.has_next)
-        stats_pages = stats_pages.get_next()
-        stats_page = stats_pages.current
-
-        self.assertEqual(1, len(stats_page), "Expected 1 record")
-        self.assertEqual(70, stats_page[0].inbound.all.all.count, "Expected 70 messages")
-
-        self.assertFalse(stats_pages.has_next)
-        stats_pages = stats_pages.get_next()
-        self.assertIsNone(stats_pages, "Expected None")
-
-    def test_app_stats_13_pagination_backwards_get_first(self):
-        params = {
-            'direction': 'backwards',
-            'start': TestRestAppStats.test_start,
-            'end': TestRestAppStats.interval_end,
-            'limit': 1,
-        }
-        stats_pages = TestRestAppStats.ably.stats(**params)
-        stats_page = stats_pages.current
-
-        self.assertEqual(1, len(stats_page), "Expected 1 record")
-        self.assertEqual(70, stats_page[0].inbound.all.all.count, "Expected 70 messages")
-
-        self.assertTrue(stats_pages.has_next)
-        stats_pages = stats_pages.get_next()
-        stats_page = stats_pages.current
-
-        self.assertEqual(1, len(stats_page), "Expected 1 record")
-        self.assertEqual(60, stats_page[0].inbound.all.all.count, "Expected 60 messages")
-
-        self.assertTrue(stats_pages.has_first)
-        stats_pages = stats_pages.get_first()
-        stats_page = stats_pages.current
-
-        self.assertEqual(1, len(stats_page), "Expected 1 record")
-        self.assertEqual(70, stats_page[0].inbound.all.all.count, "Expected 70 messages")
-
-    def test_app_stats_14_pagination_forwards_get_first(self):
-        params = {
-            'direction': 'forwards',
-            'start': TestRestAppStats.test_start,
-            'end': TestRestAppStats.interval_end,
-            'limit': 1,
-        }
-        stats_pages = TestRestAppStats.ably.stats(**params)
-        stats_page = stats_pages.current
-
-        self.assertEqual(1, len(stats_page), "Expected 1 record")
-        self.assertEqual(50, stats_page[0].inbound.all.all.count, "Expected 50 messages")
-
-        self.assertTrue(stats_pages.has_next)
-        stats_pages = stats_pages.get_next()
-        stats_page = stats_pages.current
-
-        self.assertEqual(1, len(stats_page), "Expected 1 record")
-        self.assertEqual(60, stats_page[0].inbound.all.all.count, "Expected 60 messages")
-
-        self.assertTrue(stats_pages.has_first)
-        stats_pages = stats_pages.get_first()
-        stats_page = stats_pages.current
-
-        self.assertEqual(1, len(stats_page), "Expected 1 record")
-        self.assertEqual(50, stats_page[0].inbound.all.all.count, "Expected 50 messages")
```

Comment:
It seems you've just stripped out all the tests, which will certainly make it pass ;)  Why did you do that?